File: packages/typescript/src/catalogs/sat/cfdi_4/meses.ts
import { loadCatalogObject } from '../../../utils/catalog-loader';
import type { CatalogItem } from '../../../types';

export class Meses {
  private static _data: CatalogItem[] | null = null;

  private static getData(): CatalogItem[] {
    if (this._data === null) {
      this._data = loadCatalogObject<any>('sat/cfdi_4.0/c_Meses.json').map(
        (item: any) => ({
          id: item.valor,
          name: item.descripcion || item.valor,
          ...item,
        })
      );
    }
    return this._data;
  }

  static getById(id: string): CatalogItem | undefined {
    return this.getData().find(item => item.id === id);
  }

  static isValid(id: string): boolean {
    return !!this.getById(id);
  }
}

File: packages/python/catalogmx/catalogs/sat/cfdi_4/meses.py
"""Catálogo de Meses (SAT)"""

from catalogmx.catalogs.sat.cfdi_4._resolver_views import value_rows


class Meses:
    _data = None

    @classmethod
    def _load_data(cls):
        if cls._data is None:
            rows = value_rows("cfdi_40_meses")
            cls._data = {item["valor"]: item for item in rows}
        return cls._data

    @classmethod
    def get_data(cls):
        return cls._load_data()

    @classmethod
    def get_by_id(cls, mes_id):
        """Busca un mes por su ID (e.g., '01', '02')."""
        data = cls.get_data()
        return data.get(mes_id)

    @classmethod
    def is_valid(cls, mes_id):
        """Verifica si un ID de mes es válido."""
        return mes_id in cls.get_data()
